### ISSUE TITLE
block-scoped-var false positive on renaming imported variables

### ISSUE THREAD

**Issue by vasa-chi** [eslint/eslint#2087]
test.js
```javascript
import {x as y} from "./other.js";

/* global doStuff */
doStuff(y);
```

Error:
```
test.js
  1:8   error  "x" used outside of binding context  block-scoped-var
  1:13  error  "y" used outside of binding context  block-scoped-var
  4:8   error  "y" used outside of binding context  block-scoped-var

✖ 3 problems (3 errors, 0 warnings)
```

.eslintrc
```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "block-scoped-var": 2
  },
  "ecmaFeatures": {
    "destructuring": true,
    "modules": true
  }
}
```

copy from https://github.com/eslint/eslint/issues/2059#issuecomment-82218460

**Comment by nzakas** [comment]
This rule is just a nightmare. :) Might need to look at refactoring it somehow.

**Comment by justmoon** [comment]
Seems to fail also with a plain import (no renaming):

``` js
import Network from "js/network";
```

```
test.js
  1:7  error  "Network" used outside of binding context  block-scoped-var

✖ 1 problem (1 error, 0 warnings)
```

**Comment by vslinko** [comment]
Yes, this rule doesn't work with import.

**Comment by nzakas** [comment]
Working on this.

**Comment by DenisIzmaylov** [comment]
Not at all :) It still causes the same errors in this case:
```
export createRedux from './createRedux';
...
```
Which generates follow output:
```
src/index.js
  2:7  error  "createRedux" used outside of binding context         block-scoped-var
...
```

**Comment by gaearon** [comment]
AFAIK TC39 decided not to support this syntax after all.
Does `export { default as createRedux } from './createRedux` work as intended?

**Comment by DenisIzmaylov** [comment]
@gaearon it doesn't work too:
```javascript
export {default as createRedux} from './createRedux';
...
```
```
src/index.js
  2:8   error  "default" used outside of binding context             block-scoped-var
```

**Comment by nzakas** [comment]
@denis-sokolov please open a new issue

**Comment by denis-sokolov** [comment]
I think autocompletion has accidentally completed a wrong Denis.

@DenisIzmaylov, please check @nzakas post from above.

**Comment by DenisIzmaylov** [comment]
Ok, I've created new issues https://github.com/eslint/eslint/issues/2807
